container/structure-test.sh.tpl

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -ex
+
+./%{load_statement}
```

Comment:
This is a glorious PITA to get right.  You should look at the `guess_runfiles()` logic sprinkled ~everywhere. 
 However, I'd recommend against invoking the incremental loader.

tl;dr This should `docker load -i <tarball>` and depend on the `.tar` for now (see `#3` below)

... but I'd still like to see:
1. ~`container_fs_test`: which under the hood uses `container_flatten` and checks for files in the tarball (doesn't need Docker).
1. ~`container_cfg_test`: which does the same, but consumes the `config.json` emitted (again, no Docker).
1. ~`container_exec_test`: which loads the `image` into the configured `ctx.attr.driver` and executes the configured command.

Reply:
Frankly...

`#1` could be sugar around `container_flatten` and a `tar_test` rule we upstream to `pkg_tar`.
`#2` could be sugar around `container_flatten` and a `json_test` rule we upstream somewhere.

Reply:
what's the reasoning behind not wanting to do the incremental load here? I'm fine with changing it to do an explicit docker load, just trying to understand.

I agree with having a few macros to support different use cases, but I just wasn't sure which ones we really cared about at this point in time, i.e. which will actually be used. `container_flatten`ing everything before feeding into the test seems like a generally good practice, but if we're not actually using flat images with this right now it might be overkill. maybe this should happen in a separate PR? WDYT?

Reply:
I generally regard the incremental load as (slightly) more brittle, it isn't always faster (e.g. if there are lots of images in the daemon), and I doubt we'll ever try to make it "pluggable" in the same sense as `driver`.